test/api_bridge_test.rb

```diff
@@ -0,0 +1,165 @@
+    assert_equal "https://test.unl.edu/v1/items?q=text%3Awater&hl_num=3&f[]=category%7CWritings", url
+  end
+
+  test "encode_param" do
+    # no particularly unusual characters
+    param = @api.encode_param("f[]=source|UNL Special Collections")
+    assert_equal "f[]=source%7CUNL+Special+Collections", param
+    # ampersand!
+    param = @api.encode_param("f[]=repository|Archives & Special Collections")
+    assert_equal "f[]=repository%7CArchives+%26+Special+Collections", param
+    # numbers
+    param = @api.encode_param("facet[]=date.year|1867")
+    assert_equal "facet[]=date.year%7C1867", param
+  end
+
+  # get_item_by_id
+
+  test "is_url?" do
+    # definitely not
+    assert_not @api.is_url?("some string")
+    # sort of but not quite!
+    assert_not @api.is_url?("test.unl.edu")
+    # http
+    assert @api.is_url?("http://test.unl.edu")
+    # https
+    assert @api.is_url?("https://test.unl.edu")
+    # port and path
+    assert @api.is_url?("http://localhost:3000/api/v1/items")
+  end
+
+  test "params_to_hash" do
+    # make sure this hash is not altered and returns only as string keys
+    original = { "a" => 1, b: 2 }
+    cloned = @api.params_to_hash(original)
+    assert_includes cloned, "b"
+    assert_not_includes cloned, :b
+    cloned.delete("a")
+    assert_includes original, "a"
+    # TODO ActionController::Parameters object next!
+    original = ActionController::Parameters.new({ "a" => 1, b: 2 })
+    cloned = @api.params_to_hash(original)
+    assert_includes cloned, "b"
+    assert_not_includes cloned, :b
+    cloned.delete("a")
+    assert_includes original, "a"
+  end
+
+  test "prepare_options" do
+    # make sure rows does not override num when also sent in
+    # (@api created with num)
+    opts = @api.prepare_options({ "rows" => 20 })
+    assert_not_includes opts, "rows"
+    assert_equal 5, opts["num"]
+    # TODO figure out if we want rows to override num even though
+    # there is a default num set...will need to figure something out
+    # to determine if num was the default or passed in....??
+    # api = ApiBridge::Query.new("https://test.unl.edu/v1", [], {})
+    # opts = api.prepare_options({ "rows" => 20 })
+    # assert_not_includes opts, "rows"
+    # assert_equal 20, opts["num"]
+
+    # make sure to remove rails stuff
+    req_opts = ActionController::Parameters.new({
+      controller: "items", action: "index", utf8: "✓", q: "water"
+    })
+    res_opts = { "num"=>5, "sort"=>"title|asc", "facet"=>["creator.name", "date.year", "subcategory"], "q"=>"water", "start"=>0 }
+    assert_equal res_opts, @api.prepare_options(req_opts)
+  end
+
+  test "query" do
+    skip("skipping integration testing")
+  end
+
+  test "remove_rows" do
+    # TODO see commentary in prepare_options above
+    # we should perhaps prefer rows to num for backwards compatibility
+    # if they are sending it for a specific request
```

Comment:
Note: Resolved by decision to always overwrite in newer code but I'll leave the note here fyi

I think this may depend if older sites are setting `rows` or if it's in case a user manually adds it. In the former case, we should probably respect it. In the latter, I'd maybe lean towards ignoring it going forward. Perhaps a deprecation warning in the logs though to let us know someone tried?

Reply:
Yeah, I know older sites set row on the initialization, which should have been taken care of right at the beginning.  I don't think that any apps we have send query requests with specific rows, though, they all rely on the default behavior set up at the beginning.  So I'm thinking that we should be okay in terms of behavior for our sites, but I can add a log warning.

Reply:
Oh wait, there already was a deprecation warning if rows was set.
https://github.com/CDRH/orchid/pull/115/files#diff-6afeafbfc523a81fe237405f2bfa4e52R146

Reply:
Sorry, I had seen that but forgot to mention it in the message edit I added after seeing further commits. You were on top of it :smile: